ykrt/build.rs

```diff
@@ -14,8 +14,10 @@ fn main() {
     internal_dir.push("internal_ws");
     let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
     let cargo = env::var("CARGO").unwrap();
-    let rust_flags = env::var("RUSTFLAGS").unwrap_or_else(|_| String::new());
-    let tracing_kind = find_tracing_kind(&rust_flags);
+    let tracing_kind = env::var("YK_TRACING_KIND").unwrap_or_else(|_| {
```

Comment:
Why the extra env var? So as to not duplicate work?

Reply:
Because of the `--target $HOST`, `RUSTFLAGS` is no longer passed to the build script.

Reply:
Does it hurt to always call `find_tracing_kind`?

Reply:
It will panic as `-Ctracer=hw` is no longer passed through to rustc and as such `tracermode="hw"` is no longer found in the output of `rustc --print cfg $RUSTFLAGS`.

Reply:
My last question is whether we can fold `YK_XTASK` and `YK_TRACING_KIND` into one `YK_XTASK_TRACING_KIND=hw/sw`?

Reply:
For ykrustc with hardware tracing that would work as compiling build scripts with tracing support is fine, but for cg_clif compiling them with software tracing won't work. For this reason you need to pass `--target $HOST` to cargo to make it compile build scripts without `RUSTFLAGS` set. If ykrt is used as external dependency, that would mean you would need something like `RUSTFLAGS="..." YK_TRACING_KIND="sw" cg_clif/cargo.sh build --target x86_64-unknown-linux-gnu`. In this case `YK_XTASK` must not be set.

I think I can try to include the software tracer callback as a weak function though. Hopefully weak symbols can be overridden by symbols provided by dylibs. That would make the whole last commit unnecessary.

Reply:
OK. I don't fully understand, but let's see what comes of this...